addons/point_of_sale/point_of_sale.py

```diff
@@ -1075,7 +1078,7 @@ def insert_data(data_type, values):
                 # if have_to_group_by:
 
                 sale_journal_id = order.sale_journal.id
-                period = account_period_obj.find(cr, uid, context=dict(context or {}, company_id=current_company.id))[0]
+                period = account_period_obj.find(cr, uid, dt=last_order_date, context=dict(context or {}, company_id=current_company.id))[0]
```

Comment:
No need as well. In fact, you can totaly remove the line 
period = account_period_obj.find(cr, uid, context=dict(context or {}, company_id=current_company.id))[0]